### PR TITLE
Ping360 async

### PR DESCRIPTION
**Pull request by jaxxzer**

### CODE REVIEW

File: generate/templates/ping360.py.in [hunk 1]
```diff
@@ -151,8 +151,27 @@ if __name__ == "__main__":
...
+    print(p)
+
     print("full scan in %dms, %dHz" % (1000*(tend_s - tstart_s), 400/(tend_s - tstart_s)))
 
-    print(p)
+    p.control_auto_transmit(0,399,1,0)
+
+    tstart_s = time.time()
+    for x in range(400):
+        p.wait_message([definitions.PING360_DEVICE_DATA])
```
Comment (patrickelectric) [inline generate/templates/ping360.py.in]: What you want to accomplish here ? Why is there a X variable and a loop for wait message ?

File: generate/templates/ping360.py.in [hunk 2]
```diff
@@ -151,8 +151,27 @@ if __name__ == "__main__":
...
+    tend_s = time.time()
+
+    print("full scan in %dms, %dHz" % (1000*(tend_s - tstart_s), 400/(tend_s - tstart_s)))
```
Comment (patrickelectric) [inline generate/templates/ping360.py.in]: The time that you are getting is not the scan time here, it's the time for wait_message to handle all messages.

File: generate/templates/ping360.py.in (hunk 1 above)
Comment (jaxxzer) [reply generate/templates/ping360.py.in]: i'm waiting for 400 device_data messages to arrive.

File: generate/templates/ping360.py.in (hunk 2 above)
Comment (jaxxzer) [reply generate/templates/ping360.py.in]: This is the scan time. wait_message is synchronous.

File: generate/templates/ping360.py.in (hunk 1 above)
Comment (jaxxzer) [reply generate/templates/ping360.py.in]: i'll add some comments

Comment (jaxxzer) [reply generate/templates/ping360.py.in]: done?

**Pull request merged**